src/stack/array_stack.rs

```diff
@@ -0,0 +1,134 @@
+use crate::stack::Stack;
+
+/// Uses a Vector of a given capacity (16 by default)
+struct ArrayStack<T: Clone> {
+    array: Vec<Option<T>>,
+    head: Option<usize>,
```

Comment:
Why keeping a dedicated reference to the index of the head of the stack.
If I understand well, `head` is always `array[array.length - 1]`.
Having this field forces you to keep it coherent with the rest of the state, at every modification operation.

Reply:
Good point, thanks!